src/qt/consolidateunspentwizardselectdestinationpage.cpp

```diff
@@ -0,0 +1,117 @@
+#include "consolidateunspentwizardselectdestinationpage.h"
+#include "ui_consolidateunspentwizardselectdestinationpage.h"
+
+#include "util.h"
+
+ConsolidateUnspentWizardSelectDestinationPage::ConsolidateUnspentWizardSelectDestinationPage(QWidget *parent) :
+    QWizardPage(parent),
+    ui(new Ui::ConsolidateUnspentWizardSelectDestinationPage)
+{
+    ui->setupUi(this);
+
+    ui->addressTableWidget->setSelectionMode(QAbstractItemView::SingleSelection);
+
+    // destination address selection
+    connect(ui->addressTableWidget, SIGNAL(itemSelectionChanged()), this, SLOT(addressSelectionChanged()));
+
+    ui->isCompleteCheckBox->hide();
+
+    // This is to provide a convenient way to populate the fields shown on the last page ("send" screen).
+    registerField("selectedAddressLabelField", ui->selectedAddressLabel, "text", "updateFieldsSignal()");
+    registerField("selectedAddressField", ui->selectedAddress, "text", "updateFieldsSignal()");
+
+    //This is used to control the disable/enable of the next button on this page.
+    registerField("isCompleteSelectDestination*", ui->isCompleteCheckBox);
+}
+
+ConsolidateUnspentWizardSelectDestinationPage::~ConsolidateUnspentWizardSelectDestinationPage()
+{
+    delete ui;
+}
+
+// ----------------------------------------------------------------------------- address - label
+void ConsolidateUnspentWizardSelectDestinationPage::SetAddressList(const std::map<QString, QString> addressList)
+{
+    ui->addressTableWidget->setSortingEnabled(false);
+
+    ui->addressTableWidget->clearContents();
```

Comment:
```suggestion
    ui->addressTableWidget->setRowCount(0);
```

The `clearContents()` method seems to erase the text, but the rows remain and continue to build up when pressing "back" to change the output selection. Setting the row count to zero appears to remove the actual rows.

It may be more idiomatic to resize the table outside the loop (omit `insertRow()`) and update the content with `setItem()` as it does now.

```suggestion
    ui->addressTableWidget->setRowCount(addressList.size());
```